plugins/woocommerce/src/Utilities/OrderUtil.php

```diff
@@ -0,0 +1,80 @@
+<?php
+/**
+ * A class of utilities for dealing with orders.
+ */
+
+namespace Automattic\WooCommerce\Utilities;
+
+use Automattic\WooCommerce\Internal\DataStores\Orders\CustomOrdersTableController;
+use WC_Order;
+use WP_Post;
+
+/**
+ * A class of utilities for dealing with orders.
+ */
+final class OrderUtil {
+
+	/**
+	 * Helper function to get screen name of orders page in wp-admin.
+	 *
+	 * @return string
+	 */
+	public static function get_order_admin_screen() : string {
+		return self::is_cot_enabled()
+			? wc_get_page_screen_id( 'shop-order' )
+			: 'shop_order';
+	}
+
+	/**
+	 * Helper function to get whether custom order tables are enabled or not.
+	 *
+	 * @return bool
+	 */
+	public static function is_cot_enabled() : bool {
+		return wc_get_container()->get( CustomOrdersTableController::class )->custom_orders_table_usage_is_enabled();
+	}
+
+	/**
+	 * Gets value of a meta key from WC_Data object if passed, otherwise from the post object.
+	 * This helper function support backward compatibility for meta box functions, when moving from posts based store to custom tables.
+	 *
+	 * @param WP_Post|null  $post Post object, meta will be fetched from this only when `$data` is not passed.
+	 * @param \WC_Data|null $data WC_Data object, will be preferred over post object when passed.
+	 * @param string        $key Key to fetch metadata for.
+	 * @param bool          $single Whether metadata is single.
+	 *
+	 * @return array|mixed|string Value of the meta key.
+	 */
+	public static function get_post_or_object_meta( ?WP_Post $post, ?\WC_Data $data, string $key, bool $single ) {
+		if ( isset( $data ) ) {
+			if ( method_exists( $data, "get$key" ) ) {
+				return $data->{"get$key"}();
+			}
+			return $data->get_meta( $key, $single );
+		} else {
+			return get_post_meta( $post->ID, $key, $single );
+		}
+	}
+
+
+	/**
+	 * Helper function to initialize the global $theorder object, mostly used during order meta boxes rendering.
+	 *
+	 * @param WC_Order|WP_Post $post_or_order_object Post or order object.
+	 *
+	 * @return WC_Order WC_Order object.
+	 */
+	public static function init_theorder_object( $post_or_order_object ) : WC_Order {
+		global $theorder;
```

Comment:
This is not a new object, if you see the usages, this object is already defined by the callee function. This method only helps in initialising by either a Post or an Order object, but when it's called `global $theorder` is already being used.

Reply:
Ok, I get it. But anyway, global objects still make things difficult to test.

So I wonder if we could kind of "soft deprecate" the global variable by encapsulating it in a pair of `set_current_order` and `get_current_order` methods, and start using those ourselves (instead of directly the global object) for any new code, thus encouraging others to do the same.

In that case, this method can stay as is, but perhpas it could be renamed to `init_current_order`.